File: rules/cardinality/raw-path-attribute.go
package handler

import (
	"net/http"

	"go.opentelemetry.io/otel/attribute"
)

func recordBad(r *http.Request) {
	// ruleid: tqa-card-raw-path-attribute-go
	attribute.String("http.target", r.URL.Path)

	// ruleid: tqa-card-raw-path-attribute-go
	attribute.String("http.uri", r.RequestURI)
}

func recordGood(routePattern string) {
	// ok: tqa-card-raw-path-attribute-go
	attribute.String("http.route", routePattern)
}

File: rules/cardinality/raw-path-attribute.py
def record_bad(span, request):
    # ruleid: tqa-card-raw-path-attribute-py
    span.set_attribute("http.target", request.path)

    # ruleid: tqa-card-raw-path-attribute-py
    span.set_attribute("http.uri", request.full_path)


def record_good(span, route_pattern):
    # ok: tqa-card-raw-path-attribute-py
    span.set_attribute("http.route", route_pattern)